### PR TITLE
Auto update

### PR DESCRIPTION
**Pull request by Izak88**
Added feature for user notification, when new version of the application is available.

### CODE REVIEW

File: src/electron.ts [hunk 1]
```diff
@@ -29,10 +32,35 @@ function createWindow(): Electron.BrowserWindow {
   return win;
 }
 
-app.on('ready', () => {
-  menu();
-  createWindow();
+var httpOptions = {
```
Comment (jkuri) [inline src/electron.ts]: please don't use `var` anymore. use `const` or `let` instead.

File: src/electron.ts [hunk 2]
```diff
@@ -29,10 +32,35 @@ function createWindow(): Electron.BrowserWindow {
...
+  host: 'dl.bleenco.io',
+  port: 12000,
+  path: '/update'
+};
 
+http.get(httpOptions, function(resp){
```
Comment (jkuri) [inline src/electron.ts]: can you switch this function to be inside `helpers.ts` file and export it there, I think this function shouldn't be directly in main `electron.ts` file.

File: webpack/helpers.js [hunk 3]
```diff
@@ -4,6 +4,31 @@ function root(path) {
...
+    let body = '';
+    resp.on('data', function(chunk){
+      body += chunk;
+    });
+    resp.on('end', function(){
+      let parsed = JSON.parse(body);
+      if (semver.gt(parsed.message, appVersion)){
+        notifier.notify({
+          'title': `New version is available!`,
+          'message': `bterm version ${parsed.message} is available.`
```
Comment (irmana) [inline webpack/helpers.js]: Would be better if there would be an attribute `version` with value `parsed.message`. Then you decide on the front end what kind of message you display.

File: webpack/helpers.js [hunk 4]
```diff
@@ -4,6 +4,31 @@ function root(path) {
...
+        });
+      }
+    });
+  }).on("error", function(e){
+    console.log("Got error: " + e.message);
```
Comment (irmana) [inline webpack/helpers.js]: please cleanup console log for the production, there has to be logging system in i.e. files...

File: src/electron.ts [hunk 5]
```diff
@@ -29,10 +30,19 @@ function createWindow(): Electron.BrowserWindow {
   return win;
 }
 
-app.on('ready', () => {
-  menu();
-  createWindow();
+let httpOptions = {
```
Comment (irmana) [inline src/electron.ts]: think of a service providing configuration

**Pull request merged**